setup.py

```diff
@@ -57,6 +57,7 @@
             "types-PyYAML>=6.0.0",
             "torch>=1.9.1,!=1.12.0",  # bug in 1.12.0 (https://github.com/pytorch/pytorch/pull/80345)
             "nbmake>=1.1",
+            "scikit-learn",
```

Comment:
Quick comment about dependencies: we don't like add dependency to the lib (even dev ones). Try to see if there is other way to implement the test (https://pytorch.org/docs/stable/generated/torch.pca_lowrank.html for instance ?)

Reply:
I changed it to use numpy in the test instead